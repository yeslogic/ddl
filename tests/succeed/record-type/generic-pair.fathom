let fst = fun (A: Type) (B: Type) (p: {x: A, y: B}) => p.x;
let snd = fun (A: Type) (B: Type) (p: {x: A, y: B}) => p.y;

// Regression tests for https://github.com/yeslogic/fathom/issues/507
let pair_make = fun (A: Type) (B: Type) (x: A) (y: B)     => {x = x, y = y};
let pair_id   = fun (A: Type) (B: Type) (p: {x: A, y: B}) => {x = p.x, y = p.y};
let pair_swap = fun (A: Type) (B: Type) (p: {x: A, y: B}) => {x = p.y, y = p.x};
{}
