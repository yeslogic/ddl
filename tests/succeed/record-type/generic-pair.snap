stdout = '''
let fst : fun (A : Type) (B : Type) -> { x : A, y : B } -> A = fun A B p => p.x;
let snd : fun (A : Type) (B : Type) -> { x : A, y : B } -> B = fun A B p => p.y;
() : ()
'''
stderr = ''
